src/firebolt/async_db/_types.py

```diff
@@ -219,6 +223,13 @@ def parse_type(raw_type: str) -> Union[type, ARRAY, DECIMAL]:  # noqa: C901
         return str
 
 
+BYTEA_PREFIX = "\\x"
+
+
+def _parse_bytea(str_value: str) -> bytes:
+    return bytes.fromhex(str_value[len(BYTEA_PREFIX) :])
```

Comment:
My idea was that it might be a pretty heavy operation on a large scale (since string slice is basically copy). Yet, I think we should be fine in a general case, we have less performant pieces anyway